### PR TITLE
Make a JPMS-compatible JavaNetCookieJar

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: settings.gradle.kts [hunk 1]
```diff
@@ -25,15 +25,15 @@ include(":okcurl")
...
 include(":okhttp-dnsoverhttps")
 include(":okhttp-hpacktests")
 include(":okhttp-idna-mapping-table")
+include(":okhttp-java-net-cookiejar")
 include(":okhttp-logging-interceptor")
-project(":okhttp-logging-interceptor").name = "logging-interceptor"
 include(":okhttp-sse")
 include(":okhttp-testing-support")
 include(":okhttp-tls")
-include(":okhttp-coroutines")
```
Comment (swankjesse) [inline settings.gradle.kts:36]: This list is just sorted.

**Comment by swankjesse** [comment]
I created a new PR with these changes and a few small naming changes. 
https://github.com/square/okhttp/pull/8074

I also removed the javanet.authenticator submodule; it no longer has utility! I added a deprecation note on the original class to explain why.

Thanks for driving this!

**Review by oldergod — APPROVED** [review]

**Comment by yschimke** [comment]
Legit failure?

https://github.com/square/okhttp/actions/runs/6615022089/job/17966270466#step:5:489

**Review by yschimke — APPROVED** [review]

**Comment by swankjesse** [comment]
Legit failure! I think the new commit fixes.

**Pull request merged**